File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/jetstream/kv/KeyValueStoreHolder.java
package io.micronaut.nats.docs.jetstream.kv;


import io.micronaut.context.annotation.Requires;
// tag:imports[]
import io.micronaut.nats.jetstream.annotation.KeyValueStore;
import io.nats.client.JetStreamApiException;
import io.nats.client.KeyValue;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;

import java.io.IOException;
// end::imports[]

@Requires(property = "spec.name", value = "KeyValueTest")
// tag::clazz[]
@Singleton
public class KeyValueStoreHolder {

    @Inject
    @KeyValueStore("examplebucket")
    KeyValue store; // <1>

    public void put(String key, String value) throws JetStreamApiException, IOException {
        store.put(key, value);
    }

}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/jetstream/kv/KeyValueStoreHolder.kt
package io.micronaut.nats.docs.jetstream.kv

import io.micronaut.context.annotation.Requires

// tag::imports[]
import io.micronaut.nats.jetstream.annotation.KeyValueStore
import io.nats.client.KeyValue
import jakarta.inject.Inject
import jakarta.inject.Singleton
// end::imports[]

@Requires(property = "spec.name", value = "KeyValueSpec")
// tag::clazz[]
@Singleton
class KeyValueStoreHolder {

    @Inject
    @field:KeyValueStore("examplebucket")
    lateinit var store: KeyValue // <1>

    fun put(key: String, value:String) {
        store.put(key, value)
    }

}
// end::clazz[]
